### PR TITLE
JS: Add query for measuring call graph quality

### PR DESCRIPTION
**Pull request by asger-semmle**
See https://git.semmle.com/Semmle/dist-compare/pull/134 (internal link).

### CODE REVIEW

File: javascript/ql/src/meta/analysis-quality/ResolvedCalls.ql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * @name Resolved calls
+ * @description The number of calls that could be resolved to its target.
```
Comment (xiemaisi) [inline javascript/ql/src/meta/analysis-quality/ResolvedCalls.ql]:
```suggestion
 * @description The number of calls that could be resolved to a target.
```

File: javascript/ql/src/meta/analysis-quality/UnresolvedCalls.ql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * @name Unresolved calls
+ * @description The number of calls that could not be resolved to its target.
```
Comment (xiemaisi) [inline javascript/ql/src/meta/analysis-quality/UnresolvedCalls.ql]:
```suggestion
 * @description The number of calls that could not be resolved to a target.
```

File: javascript/ql/src/meta/analysis-quality/CallGraphQuality.qll [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+}
+
+/**
+ * Gets a node that was derived from an import of `moduleName`.
+ *
+ * This is a rough approximation as it follows all property reads, invocations,
+ * and callbacks, so some of these might refer to internal objects.
+ *
+ * Additionally, we don't recognize when a project imports another file in the
+ * same project using it module name (e.g. import "vscode" from inside the vscode project).
```
Comment (xiemaisi) [inline javascript/ql/src/meta/analysis-quality/CallGraphQuality.qll]:
```suggestion
 * same project using it module name (for example, import "vscode" from inside the vscode project).
```

File: javascript/ql/src/meta/analysis-quality/ResolvedCallCandidates.ql [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/**
+ * @name Resolvable call site candidates
```
[the remark this answers is not in the document]
Comment (asger-semmle) [reply javascript/ql/src/meta/analysis-quality/ResolvedCallCandidates.ql]: Renamed files, IDs, and the `ResolvedCall` and `UnresolvedCall` classes to use "resolvable" consistently.

**Pull request merged**